### ISSUE TITLE
Benchmark in HTTP/2 and H2Load

### ISSUE THREAD

**Issue by proyb6** [the-benchmarker/web-frameworks#4367]
I think it is worth to discuss if there is any interest in benchmark with http/2 since it has been widely used across the web ecosystem.

**Comment by waghanza** [comment]
sure it will be awesome to gather any feedback on this ... probably better to use **Discussions** tab for this

I'm 100% ok for this, but have to be properly planned in a roadmap (btw I'm not sure `wrk` could support http2)

**Comment by proyb6** [comment]
wrk does not support H2, and all use h2load to benchmark among reverse proxy like LiteSpeed, Caddy, Nginx, etc.

**Comment by waghanza** [comment]
who is all ?

better to have a discussion about sieger, then.

could any like be https://github.com/tsenart/vegeta

either one for http1 / http2 / https or several tools

**Comment by proyb6** [comment]
Centminmod, LiteSpeed, etc use h2load.

https://community.centminmod.com/threads/caddy-http-2-server-benchmarks-part-2.12873/
https://vpnwp.com/performing-performance-request-test-between-nginx-vs-openlitespeed-with-h2load.wp

Have not use Vegeta or alternative, but it would be nice to reopen the old discussion on benchmark tools.

**Comment by schamberg97** [comment]
This generally doesn't make sense for node.js as the most common pattern is to place node.js instances behind reverse proxy like nginx. Moreover, nginx doesn't use http2 to contact backend servers, relying on http 1.1. Http 2 is only implemented for client requests

**Comment by waghanza** [comment]
> nginx doesn't use http2 to contact backend servers

are you sure that nginx could not use http2 to proxify any http2 requests 

> http 2 is only implemented for client requests

this is generally the big part on network communication

> Thus, in many cases http2 support in framework might be irrelevant at this point

Not sure about this, but in deed when behind a proxy on http1.1 or behind on app engin that does not support http2

**Comment by schamberg97** [comment]
Absolutely, nginx have never implemented the ability to proxy http2 requests
